src/main/java/org/jpeek/FileTarget.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.jpeek;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+
+/**
+ * File Target.
+ * @author Felipe Moreno (oridan@gmail.com)
+ * @version $Id$
+ * @since 1.0
+ */
+public final class FileTarget implements Target {
+
+    /**
+     * The file target.
+     */
+    private final File target;
+
+    /**
+     * Overwrites if true.
+     */
+    private final Boolean overwrite;
+
+    /**
+     * Ctor.
+     * @param target Target dir
+     * @param overwrite Overwrite if exists
+     */
+    public FileTarget(final File target, final Boolean overwrite) {
+        this.target = target;
+        this.overwrite = overwrite;
+    }
+
+    @Override
+    public Path toPath() throws IOException {
+        if (this.target.exists()) {
+            if (this.overwrite) {
+                deleteDir(this.target);
+            } else {
+                throw new IllegalStateException(
+                    String.format(
+                        "Directory/file already exists: %s",
+                        this.target.getAbsolutePath()
```

Comment:
@oridan I think this message does not represents what is really happening: looks like that this exception will be thrown when that `ovewrite` flag is not set, meaning that the path or file cannot be overwritten, not that it already exists; please correct this message then.

Reply:
@oridan So, I think the message should inform these two conditions, WDYT?

Reply:
@paulodamaso Yes, the message could inform these two conditions. I just submitted the changes, check it out.